core/src/fgm/fgj/gamejamgame/GameEvent.java

```diff
@@ -25,128 +28,156 @@ public boolean isPositive(){
 
 	public GameEvent(SolarSystem solarSystem, int fuelCost, Planet planet, Ship ship) {
 		this.eventText = "Nothing happened.";
+		this.isPositive = true;
 		if (ship == null) {
 			throw new IllegalArgumentException("A game event cannot occur with a null ship.");
 		}
+		this.ship = ship;
 		if(solarSystem == null && planet == null){
 			this.eventContext = EventContext.GALAXY;
 			ship.getCargoBay().decreaseFuel(1);
 		}else if(planet == null){
+			this.solarSystem = solarSystem;
 			this.eventContext = EventContext.SOLAR_SYSTEM;
 			if(fuelCost > 0){
 				ship.getCargoBay().decreaseFuel(fuelCost);
 			}else{
 				throw new IllegalArgumentException("Fuel costs to visit another solar system cannot be less than 1.");
 			}
 		}else{
+			this.solarSystem = solarSystem;
+			this.planet = planet;
 			this.eventContext = EventContext.PLANET;
 			ship.getCargoBay().decreaseFuel(1);
 		}
+
```

Comment:
We still need the fuel check to force a losing game state. This is also true for atmosphere gas.